src/CoreWCF.Primitives/src/CoreWCF/Channels/ChannelDemuxer.cs

```diff
@@ -0,0 +1,3848 @@
+    //                    MessageTransmitTraceRecord.CreateReceiveTraceRecord(message), this, null);
+    //            }
+    //        }
+    //    }
+
+    //    class DuplexChannelWrapper : DuplexChannel
+    //    {
+    //        IDuplexChannel innerChannel;
+
+    //        public DuplexChannelWrapper(ChannelManagerBase channelManager, IDuplexChannel innerChannel)
+    //            : base(channelManager, innerChannel.LocalAddress)
+    //        {
+    //            this.innerChannel = innerChannel;
+    //        }
+
+    //        public override EndpointAddress RemoteAddress
+    //        {
+    //            get { return this.innerChannel.RemoteAddress; }
+    //        }
+
+    //        public override Uri Via
+    //        {
+    //            get { return this.innerChannel.Via; }
+    //        }
+
+    //        protected override void OnSend(Message message, TimeSpan timeout)
+    //        {
+    //            this.innerChannel.Send(message, timeout);
+    //        }
+
+    //        protected override IAsyncResult OnBeginSend(Message message, TimeSpan timeout, AsyncCallback callback, object state)
+    //        {
+    //            return this.innerChannel.BeginSend(message, timeout, callback, state);
+    //        }
+
+    //        protected override void OnEndSend(IAsyncResult result)
+    //        {
+    //            this.innerChannel.EndSend(result);
+    //        }
+
+    //        protected override IAsyncResult OnBeginOpen(TimeSpan timeout, AsyncCallback callback, object state)
+    //        {
+    //            return new CompletedAsyncResult(callback, state);
+    //        }
+
+    //        protected override void OnEndOpen(IAsyncResult result)
+    //        {
+    //            CompletedAsyncResult.End(result);
+    //        }
+
+    //        protected override void OnOpen(TimeSpan timeout)
+    //        {
+    //        }
+    //    }
+    //}
+
+    class ReplyChannelDemuxer : DatagramChannelDemuxer<IReplyChannel, RequestContext>
+    {
+        private static IList<Type> s_supportedChannelTypes = new List<Type> { typeof(IReplyChannel) };
+
+        public override Uri BaseAddress => throw new NotImplementedException();
+
+        public override Binding Binding => throw new NotImplementedException();
+
+        public override IList<Type> SupportedChannelTypes => s_supportedChannelTypes;
+
+        public ReplyChannelDemuxer()
+        {
+        }
+
+        public override Task<IServiceChannelDispatcher> CreateServiceChannelDispatcherAsync(IChannel channel)
+        {
+            return Task.FromResult<IServiceChannelDispatcher>(new ReplyChannelDispatcher(this, channel));
+        }
+
+        protected override void AbortItem(RequestContext request)
+        {
+            AbortMessage(request);
+            request.Abort();
+        }
+
+        protected override void EndpointNotFound(RequestContext request)
+        {
+            //bool abortItem = true;
```

Comment:
Is this method ever called? We need to make sure that we handle the scenario where the demuxer can't find a matching dispatcher. This is also a test gap. Can you open an issue to test the scenario where a message is received for a session we don't know about. This actually happens with a misconfigured load balancer which doesn't use session affinity.

Reply:
https://github.com/CoreWCF/CoreWCF/pull/172#discussion_r523249734